### PR TITLE
Adding AWS Lambda middleware ChatGPT & Redshift ChatGPT Actions cookbook

### PR DESCRIPTION
**Pull request by pap-openai**
## Summary

- Adding Redshift to the GPT Actions Library
- Adding AWS Middleware GPT Actions Library

## Motivation

Helping our customers that want to connect ChatGPT to Redshift
Helping our customers that want to connect AWS Services to ChatGPT or use Lambda to do pre-processing

---

## For new content

When contributing new content, read through our [contribution guidelines](https://github.com/openai/openai-cookbook/blob/main/CONTRIBUTING.md), and mark the following action items as completed:

- [x] I have added a new entry in [registry.yaml](https://github.com/openai/openai-cookbook/blob/main/registry.yaml) (and, optionally, in [authors.yaml](https://github.com/openai/openai-cookbook/blob/main/authors.yaml)) so that my content renders on the cookbook website.
- [x] I have conducted a self-review of my content based on the [contribution guidelines](https://github.com/openai/openai-cookbook/blob/main/CONTRIBUTING.md#rubric):
  - [x] Relevance: This content is related to building with OpenAI technologies and is useful to others.
  - [x] Uniqueness: I have searched for related examples in the OpenAI Cookbook, and verified that my content offers new insights or unique information compared to existing documentation.
  - [x] Spelling and Grammar: I have checked for spelling or grammatical mistakes.
  - [x] Clarity: I have done a final read-through and verified that my submission is well-organized and easy to understand.
  - [x] Correctness: The information I include is correct and all of my code executes successfully.
  - [x] Completeness: I have explained everything fully, including all necessary references and citations.

We will rate each of these areas on a scale from 1 to 4, and will only accept contributions that score 3 or higher on all areas. Refer to our [contribution guidelines](https://github.com/openai/openai-cookbook/blob/main/CONTRIBUTING.md) for more details.

### CODE REVIEW

File: images/chatgpt/aws_lambda_1.png
Comment (aaronwilkowitz-openai) [inline images/chatgpt/aws_lambda_1.png:1]: Is there middleware function name sensitive? I'm assuming not but just double checking

File: images/chatgpt/aws_lambda_15.png
Comment (aaronwilkowitz-openai) [inline images/chatgpt/aws_lambda_15.png:1]: Is client ID sensitive?

File: images/chatgpt/aws_lambda_17.png
Comment (aaronwilkowitz-openai) [inline images/chatgpt/aws_lambda_17.png:1]: Let's black out the callback URL

File: images/chatgpt/aws_lambda_18.png
Comment (aaronwilkowitz-openai) [inline images/chatgpt/aws_lambda_18.png:1]: Same here - black out callback URL

File: examples/chatgpt/gpt_actions_library/gpt_middleware_aws_function.ipynb [hunk 1]
```diff
@@ -0,0 +1,421 @@
...
+   "source": [
+    "## Create AWS Lambda Function"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "To create an AWS Function you can use AWS SAM. An example of a SAM Template can be found [here](https://github.com/openai/redshift-middleware/blob/main/template.yaml) [0].\n",
```
Comment (aaronwilkowitz-openai) [inline examples/chatgpt/gpt_actions_library/gpt_middleware_aws_function.ipynb:91]: This is not a publicly accessible repo - should it be? Or should you remove the URL?

**Review by aaronwilkowitz-openai — COMMENTED** [review]
Looks really good - made a few small comments

File: images/chatgpt/aws_lambda_1.png
Comment (pap-openai) [reply images/chatgpt/aws_lambda_1.png:1]: nop, but assuming it's better to blacking it out regardless, edited! ty

File: images/chatgpt/aws_lambda_15.png
Comment (pap-openai) [reply images/chatgpt/aws_lambda_15.png:1]: nop (although best practice to not share it in the open but it's often on front-end accessible by public), but assuming it's better to blacking it out regardless, edited! ty

File: images/chatgpt/aws_lambda_17.png
Comment (pap-openai) [reply images/chatgpt/aws_lambda_17.png:1]: done! (url is also not used anymore as this GPT has been destroyed :))

File: images/chatgpt/aws_lambda_18.png
Comment (pap-openai) [reply images/chatgpt/aws_lambda_18.png:1]: done! (url is also not used anymore as this GPT has been destroyed :))

File: examples/chatgpt/gpt_actions_library/gpt_middleware_aws_function.ipynb (hunk 1 above)
Comment (pap-openai) [reply examples/chatgpt/gpt_actions_library/gpt_middleware_aws_function.ipynb]: good catch thanks!

**Review by aaronwilkowitz-openai — APPROVED** [review]
Looks great!

**Pull request merged**